frameworks/Rust/faf/src/main.rs

```diff
@@ -35,55 +30,43 @@ const PLAINTEXT_BASE: &[u8] = const_concat_bytes!(
 
 const PLAINTEXT_BASE_LEN: usize = const_len(PLAINTEXT_BASE);
 
+const PLAINTEXT_TEST: &[u8] = b"HTTP/1.1 200 OK\r\nServer: F\r\nContent-Type: text/plain\r\nContent-Length: 13\r\nDate: Thu, 18 Nov 2021 23:15:07 GMT\r\n\r\nHello, World!";
+const PLAINTEXT_TEST_LEN: usize = const_len(PLAINTEXT_TEST);
+
 #[inline]
 fn cb(
-   method: *const i8,
+   method: *const u8,
    method_len: usize,
-   path: *const i8,
+   path: *const u8,
    path_len: usize,
-   _headers: &[faf::phr_header; MAX_HEADERS_TO_PARSE],
-   _num_headers: usize,
    response_buffer: *mut u8,
+   date_buff: *const u8,
 ) -> usize {
    unsafe {
       if likely(method_len >= GET_LEN && path_len >= ROUTE_PLAINTEXT_LEN) {
-         if likely(memcmp(GET.as_ptr() as *const i8, method, GET_LEN) == 0) {
+         if likely(memcmp(GET.as_ptr(), method, GET_LEN) == 0) {
             // For performance purposes, this will successfully match '/p' to '/plaintext' and '/pickle'. Use with caution
```

Comment:
I've missed this before, and I'm not sure how many other people are doing it, but I don't like this. I want it to only match `/plaintext` but we're not enforcing that.

Reply:
I updated the code to perform a 'full' comparison. It should be good to go now